### ISSUE TITLE
Failed when trying to install pythonnet for a new project

### ISSUE THREAD

**Issue by SherkeyXD** [beeware/briefcase#557]
**Describe the bug**
Recently I tried to generate a new project using briefcase, but when I use the command `briefcase new` it went wrong. 

**To Reproduce**
1. Generate a new project(in venv). 
2. Use `briefcase new`
3. See error

**Expected behavior**
Actually, it works well on Win7, but both of my computer are on Win10 and have the same problem. 

**Screenshots**
I have a log down here. 

**Environment:**
 - Operating System:Windows 10 20h2
 - Python version:3.9.0
 - Software versions:
   - Briefcase:newest
   - Toga:newest
   - ...

**Additional context**
Other parts all works well, except the pythonnet.
I am actually a Chinese so I am not doing so good in English, and there are a little chinese in the log.
Here is the log.
```
> (venv) PS E:\VSCodeFiles\Python\zhiInquiry-GUI\zhiInquiry-GUI> pip install pythonnet
Looking in indexes: https://mirrors.aliyun.com/pypi/simple     # mirror
Collecting pythonnet
  Using cached https://mirrors.aliyun.com/pypi/packages/19/3a/058cee38730776bcecc2d3dd54fe50ac3f8c77077ba9dc10bb59e70549e2/pythonnet-2.5.1.tar.gz (1.9 MB)
Requirement already satisfied: pycparser in e:\vscodefiles\python\zhiinquiry-gui\venv\lib\site-packages (from pythonnet) (2.20)
Building wheels for collected packages: pythonnet
  Building wheel for pythonnet (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'e:\vscodefiles\python\zhiinquiry-gui\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\25329\\AppData\\Local\\Temp\\pip-install-48ji0l4o\\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\\setup.py'"'"'; __file__='"'"'C:\\Users\\25329\\AppData\\Local\\Temp\\pip-install-48ji0l4o\\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\25329\AppData\Local\Temp\pip-wheel-c4olbt7i'
       cwd: C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\
  Complete output (93 lines):
  running bdist_wheel
  running build
  running build_ext
  正在检查来自 https://www.nuget.org/api/v2/ 的更新。
  当前正在运行 NuGet.exe 4.1.0。
  正在将 NuGet.exe 更新到 5.8.1。
  更新成功。
  Cannot find the specified version of msbuild: '14'
  MSBuild auto-detection: using msbuild version '4.0' from 'C:\Windows\Microsoft.NET\Framework64\v4.0.30319'.
  C:\Users\25329\AppData\Local\Temp\NuGetScratch\znattvvl.tmn.nugetrestore.targets(455,20): error MSB4086: 尝试在条件“'$(MSBuildAssemblyVersion)' < '15.0'”中对计算结果为“”而不是数字的“$(MSBuildAssemblyVersion)”进行数值比较。 [C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\src\runtime\Python.Runtime.csproj]
  C:\Users\25329\AppData\Local\Temp\NuGetScratch\znattvvl.tmn.nugetrestore.targets(455,20): error MSB4086: 尝试在条件“'$(MSBuildAssemblyVersion)' < '15.0'”中对计算结果为“”而不是数字的“$(MSBuildAssemblyVersion)”进行数值比较。 [C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\src\testing\Python.Test.csproj]
  C:\Users\25329\AppData\Local\Temp\NuGetScratch\znattvvl.tmn.nugetrestore.targets(455,20): error MSB4086: 尝试在条件“'$(MSBuildAssemblyVersion)' < '15.0'”中对计算结果为“”而不是数字的“$(MSBuildAssemblyVersion)”进行数值比较。 [C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\src\embed_tests\Python.EmbeddingTest.csproj]
  C:\Users\25329\AppData\Local\Temp\NuGetScratch\znattvvl.tmn.nugetrestore.targets(455,20): error MSB4086: 尝试在条件“'$(MSBuildAssemblyVersion)' < '15.0'”中对计算结果为“”而不是数字的“$(MSBuildAssemblyVersion)”进行数值比较。 [C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\src\console\Console.csproj]
  C:\Users\25329\AppData\Local\Temp\NuGetScratch\znattvvl.tmn.nugetrestore.targets(455,20): error MSB4086: 尝试在条件“'$(MSBuildAssemblyVersion)' < '15.0'”中对计算结果为“”而不是数字的“$(MSBuildAssemblyVersion)”进行数值比较。 [C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\src\clrmodule\clrmodule.csproj]
  警告: Error reading msbuild project information, ensure that your input solution or project file is valid. NETCore and UAP projects will be skipped, only packages.config files will be restored.
  Restoring NuGet package UnmanagedExports.1.2.7.
  Restoring NuGet package System.ValueTuple.4.5.0.
  Restoring NuGet package NUnit.3.12.0.
  Restoring NuGet package NUnit.ConsoleRunner.3.11.1.
  Adding package 'System.ValueTuple.4.5.0' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Adding package 'UnmanagedExports.1.2.7' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Adding package 'NUnit.3.12.0' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Adding package 'NUnit.ConsoleRunner.3.11.1' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Added package 'UnmanagedExports.1.2.7' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Added package 'NUnit.ConsoleRunner.3.11.1' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Added package 'System.ValueTuple.4.5.0' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  Added package 'NUnit.3.12.0' to folder 'C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\packages'
  NuGet Config files used:
      C:\Users\25329\AppData\Roaming\NuGet\NuGet.Config
  Feeds used:
      C:\Users\25329\.nuget\packages\
      https://api.nuget.org/v3/index.json
  Installed:
      4 package(s) to packages.config projects
  Traceback (most recent call last):
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\tools\geninterop\geninterop.py", line 307, in <module>
      sys.exit(main())
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\tools\geninterop\geninterop.py", line 287, in main
      python_h = preprocess_python_headers()
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\tools\geninterop\geninterop.py", line 207, in preprocess_python_headers
      for line in _check_output(cmd).splitlines():
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\tools\geninterop\geninterop.py", line 41, in _check_output
      output = subprocess.check_output(*args, **kwargs)
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 420, in check_output
      return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 501, in run
      with Popen(*popenargs, **kwargs) as process:
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 947, in __init__
      self._execute_child(args, executable, preexec_fn, close_fds,
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 1416, in _execute_child
      hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
  FileNotFoundError: [WinError 2] 系统找不到指定的文件。
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\setup.py", line 630, in <module>
      setup(
    File "e:\vscodefiles\python\zhiinquiry-gui\venv\lib\site-packages\setuptools\__init__.py", line 165, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\setup.py", line 612, in run
      return bdist_wheel.bdist_wheel.run(self)
    File "e:\vscodefiles\python\zhiinquiry-gui\venv\lib\site-packages\wheel\bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\command\build.py", line 135, in run
      self.run_command(cmd_name)
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 340, in run
      self.build_extensions()
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 449, in build_extensions
      self._build_extensions_serial()
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 474, in _build_extensions_serial
      self.build_extension(ext)
    File "C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\setup.py", line 302, in build_extension
      subprocess.check_call([sys.executable, geninterop, interop_file])
    File "C:\Users\25329\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 373, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['e:\\vscodefiles\\python\\zhiinquiry-gui\\venv\\scripts\\python.exe', 'tools\\geninterop\\geninterop.py', 'src\\runtime\\interop39.cs']' returned non-zero exit status 1. ----------------------------------------
  ERROR: Failed building wheel for pythonnet
  Running setup.py clean for pythonnet
Failed to build pythonnet
Installing collected packages: pythonnet
    Running setup.py install for pythonnet ... error
    ERROR: Command errored out with exit status 1:
     command: 'e:\vscodefiles\python\zhiinquiry-gui\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\25329\\AppData\\Local\\Temp\\pip-install-48ji0l4o\\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\\setup.py'"'"'; __file__='"'"'C:\\Users\\25329\\AppData\\Local\\Temp\\pip-install-48ji0l4o\\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\25329\AppData\Local\Temp\pip-record-rfpe4guf\install-record.txt' --single-version-externally-managed --compile --install-headers 'e:\vscodefiles\python\zhiinquiry-gui\venv\include\site\python3.9\pythonnet'
         cwd: C:\Users\25329\AppData\Local\Temp\pip-install-48ji0l4o\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\
    Complete output (6 lines):
    usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
       or: setup.py --help [cmd1 cmd2 ...]
       or: setup.py --help-commands
       or: setup.py cmd --help
    error: option --single-version-externally-managed not recognized ----------------------------------------
ERROR: Command errored out with exit status 1: 'e:\vscodefiles\python\zhiinquiry-gui\venv\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\25329\\AppData\\Local\\Temp\\pip-install-48ji0l4o\\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\\setup.py'"'"'; __file__='"'"'C:\\Users\\25329\\AppData\\Local\\Temp\\pip-install-48ji0l4o\\pythonnet_824062e28fdd4a9a8b8949709ca04e6a\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\25329\AppData\Local\Temp\pip-record-rfpe4guf\install-record.txt' --single-version-externally-managed --compile --install-headers 'e:\vscodefiles\python\zhiinquiry-gui\venv\include\site\python3.9\pythonnet' Check the logs for full command output.
```

**Comment by freakboy3742** [comment]
Thanks for the report. However, this isn't a bug with Briefcase. The problems is that you're using Python 3.9 on Windows; Windows requires Python.net, and Python.net hasn't been released for Python 3.9. 

Python.net isn't managed by the BeeWare project, so we have no control over their release schedule. Until they release a version for Python 3.9, Briefcase won't be able to support Python 3.9 on Windows either.

That said, a version of this problem emerges every time a new Python version is released, so it would be good to catch this error and present the user with a better error message that indicates the source of the problem.

**Comment by freakboy3742** [comment]
This was reported again as #578; this time the error was slightly different, because the user had enough of a C compiler chain installed that the failure occurred during compilation.
